mzLib/Omics/Fragmentation/Oligo/DissociationTypeCollection.cs

```diff
@@ -1 +1,258 @@
            {
                {
                    FragmentationTerminus.FivePrime, new List<ProductType>
                    {
                        ProductType.a, ProductType.aWaterLoss, ProductType.aBaseLoss,
                        ProductType.b, ProductType.bWaterLoss, ProductType.bBaseLoss,
                        ProductType.c, ProductType.cWaterLoss, ProductType.cBaseLoss,
                        ProductType.d, ProductType.dWaterLoss, ProductType.dBaseLoss, 
                    }
                },
                {
                    FragmentationTerminus.ThreePrime, new List<ProductType>
                    {
                        ProductType.w, ProductType.wWaterLoss, ProductType.wBaseLoss,
                        ProductType.x, ProductType.xWaterLoss, ProductType.xBaseLoss,
                        ProductType.y, ProductType.yWaterLoss, ProductType.yBaseLoss,
                        ProductType.z, ProductType.zWaterLoss, ProductType.zBaseLoss,
                    }
                },
                {
                    FragmentationTerminus.Both, new List<ProductType>
                    {

                        ProductType.a, ProductType.aWaterLoss, ProductType.aBaseLoss,
                        ProductType.b, ProductType.bWaterLoss, ProductType.bBaseLoss,
                        ProductType.c, ProductType.cWaterLoss, ProductType.cBaseLoss,
                        ProductType.d, ProductType.dWaterLoss, ProductType.dBaseLoss, 
                        ProductType.w, ProductType.wWaterLoss, ProductType.wBaseLoss,
                        ProductType.x, ProductType.xWaterLoss, ProductType.xBaseLoss,
                        ProductType.y, ProductType.yWaterLoss, ProductType.yBaseLoss,
                        ProductType.z, ProductType.zWaterLoss, ProductType.zBaseLoss,
                        ProductType.M
                    }
                }
            };


        public static List<ProductType> GetRnaTerminusSpecificProductTypes(
            this FragmentationTerminus fragmentationTerminus)
        {
            return ProductIonTypesFromSpecifiedTerminus[fragmentationTerminus];
        }

        /// <summary>
        /// Returns all product ion types based upon specified terminus
        /// </summary>
        /// <param name="dissociationType"></param>
        /// <param name="fragmentationTerminus"></param>
        /// <returns></returns>
        public static List<ProductType> GetRnaTerminusSpecificProductTypesFromDissociation(
            this DissociationType dissociationType, FragmentationTerminus fragmentationTerminus)
        {
            var terminusSpecific = fragmentationTerminus.GetRnaTerminusSpecificProductTypes();
            var dissociationSpecific = dissociationType.GetRnaProductTypesFromDissociationType();
            return terminusSpecific.Intersect(dissociationSpecific).ToList();
        }
    }
}

```

Comment:
This is weirdly showing up as a single line. Something messed up with the carriage returns